src/Compilers/Core/Portable/Compilation/IOperationWalker.cs

```diff
@@ -0,0 +1,394 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a <see cref="IOperationVisitor"/> that descends an entire <see cref="IOperation"/> tree
+    /// visiting each IOperation and its child IOperation nodes in depth-first order.
+    /// </summary>
+    public abstract class IOperationWalker : IOperationVisitor
```

Comment:
This should be named OperationWalker... Same with IOperationVisitor, class names should not include the I.